index.js

```diff
@@ -28,6 +28,7 @@ function encodeSignature (signature, recovery, compressed, segwitType) {
   if (segwitType !== undefined) {
     recovery += 8
     if (segwitType === SEGWIT_TYPES.P2WPKH) recovery += 4
+    if (segwitType === SEGWIT_TYPES.P2SH_P2WPKH) recovery -= 4
```

Comment:
This fixes the problem, but is obviously not the correct solution.

decodeSignature is not changed, so with this "fix" a p2sh-p2wpkh signed by this commit will not be able to be verified by this commit.

Reply:
I'll add some tests which use both the sign and verify to ensure they are sound.

Reply:
I would just wait.

I want to hear some feedback from Electrum / Trezor first before deciding what to do.

Reply:
Sounds good, I'll wait. :slightly_smiling_face: